docs/index.asciidoc

```diff
@@ -415,6 +420,35 @@ deprecated[6.4.0, Replaced by <<plugins-{type}s-{plugin}-ssl_supported_protocols
 The minimum TLS version allowed for the encrypted connections.
 The value must be one of the following: 1.1 for TLS 1.1, 1.2 for TLS 1.2, 1.3 for TLS 1.3
 
+[id="plugins-{type}s-{plugin}-enrich"]
+===== `enrich`
+* Value type is <<array,array>>
+
+By default, the plugin initializes ECS-disabled plain codec, source and codec metadata such as source host or `[event][original]` will not be added.
+`enrich` option helps to manage which metadata to be included. Available options are:
+
+[cols="<,<"]
+|=======================================================================
+|Option |Description
+
+|`none`                  |Not to include any data
+|`default` alias | `codec_metadata` and `codec_metadata` will be included
+|`all` alias | All `codec_metadata`, `codec_metadata` and `ssl_peer_metadata` will be included
```

Comment:
Do we really need the `default` or `all` aliases? I would argue that we should completely drop them. So that:

1. The documentation should highlight that the default behavior (when the `enrich` option is NOT provided) is to default to `codec_metadata` and `ssl_peer_metadata`.
2. If the user wants to disable enrichment OR override the default behavior, then they can add it.

As such, having a `default` option makes no sense to me because not providing the option is already default. It would just be repetitive and IMHO, useless.


As far as the `all` alias, while I understand that we would be making it easier for users to benefit from all the metadata additions, it would also potentially be a reason why customers change the settings after upgrade if we add a new metadata option. Instead, users should be able to *opt-in* into that option if they have manually changed the `enrich` option.

Reply:
Also, it is fine to drop the "default" alias, e.g., it doesn't make sense to do:

enrich => ["default"]

Reply:
After chatting with @yaauie, and going over the PR the solution for this is to:

1. Make documentation that clearly describes what the default behavior is *when not providing the option*
2. The `default` option, should be an internal thing, not something it makes sense for users to set (e.g. enrich => default). It brings no value to have that as an option users can change. To simplify, we should only make doc changes to accordingly describe the default behavior.
3. I'm fine keeping `all` as a user option.

Reply:
Thanks for the feedback @roaksoax and it makes sense and I have updated the logic.